src/pages/index.js

```diff
@@ -67,36 +95,151 @@ addNewCardButton.addEventListener("click", () => {
   newCardPopup.open();
 });
 
-const profilePopup = new PopupWithForm("#edit-modal", (formData) => {
+//=========================
+//=== Get Profile Popup ===
+//=========================
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData);
+    userInfo.setUserAvatar(userData);
+  })
+  .catch((error) => {
+    console.error("info not fetched from the server", error);
+  });
+
+//===========================
+//=== Edit Profile Popup ====
+//===========================
+const profilePopup = new PopupWithForm("#edit-profile-modal", (formData) => {
   const name = formData["name"];
   const job = formData["description"];
+  profilePopup.setButtonText(true);
+  api
+    .updateProfile(name, job)
+    .then((updatedData) => {
+      userInfo.setUserInfo({
+        name: updatedData.name,
+        about: updatedData.about,
+      });
+      profilePopup.close();
+    })
+    .catch((error) => {
+      console.error("info not updated", error);
+    })
+    .finally(() => {
+      profilePopup.setButtonText();
+    });
+});
 
-  userInfo.setUserInfo({ title: name, description: job });
-  profilePopup.close();
+const userInfo = new UserInfo({
+  nameSelector: ".profile__title",
+  jobSelector: ".profile__description",
+  avatarSelector: ".avatar__image",
 });
 
 profilePopup.setEventListeners();
 
 profileEditButton.addEventListener("click", () => {
-  
-    const currentUserInfo = userInfo.getUserInfo();
-    nameInput.value = currentUserInfo.title;
-    jobInput.value = currentUserInfo.description;
-    editFormValidator.resetValidation();
-  
+  profilePopup.open();
+  const currentUserInfo = userInfo.getUserInfo();
+  nameInput.value = currentUserInfo.title;
+  jobInput.value = currentUserInfo.description;
+  editFormValidator.resetValidation();
+
   profilePopup.open();
```

Comment:
`  profilePopup.open();
`

You have the same code in line 144. Please, delete this one